### PR TITLE
Increase amount of space allocated for metadata

### PR DESCRIPTION
**Pull request by jbaublitz**
Closes https://github.com/stratis-storage/stratisd/issues/3192

@bleve This patch should resolve your metadata issue if you would like to test it. I've already tested both components of it individually, and it appears to resolve the problem.

### CODE REVIEW

**Comment by jbaublitz** [comment]
@mulkieran I think we could easily add an automated test to our unit tests soon to ensure we never run out of metadata space if you'd like.

File: src/engine/strat_engine/cmd.rs [hunk 1]
```diff
@@ -456,7 +456,7 @@ pub fn thin_metadata_size(
         .read_to_string(&mut output)?;
     if is_ok {
         Ok(min(
-            Sectors(
+            8u64 * Sectors(
```
Comment (mulkieran) [inline src/engine/strat_engine/cmd.rs:459]: We had better make this a constant and document it as best we can, even though there's not much to say.

File: src/engine/strat_engine/thinpool/thinpool.rs [hunk 2]
```diff
@@ -259,6 +259,17 @@ fn status_to_usage(status: Option<ThinPoolStatus>) -> Option<ThinPoolUsage> {
     })
 }
 
+/// Convert the thin pool status to the metadata low water mark.
```
Comment (mulkieran) [inline src/engine/strat_engine/thinpool/thinpool.rs:262]: This is fine (exactly what we do for ThinPoolUsage), but I'm starting to wonder why we don't just store the whole ThinPoolStatus in the thinpool implementation and extract the data we need from that when we want it. We can either tackle that in this PR, or I'll file an issue and we can work on that later.

File: src/engine/strat_engine/thinpool/thinpool.rs [hunk 3]
```diff
@@ -953,6 +979,7 @@ impl ThinPool {
         pool_uuid: PoolUuid,
         backstore: &mut Backstore,
         new_thin_limit: Option<u64>,
+        is_lowater: bool,
```
Comment (mulkieran) [inline src/engine/strat_engine/thinpool/thinpool.rs:982]: Better document this parameter as well.

**Review by mulkieran — CHANGES_REQUESTED** [review]
A couple of requests and one comment.

From the code, I conclude that stratisd was already receiving the thin metadata low water mark event, so there is literally no work to do on the event handling side?

File: src/engine/strat_engine/thinpool/thinpool.rs (hunk 3 above)
Comment (jbaublitz) [reply src/engine/strat_engine/thinpool/thinpool.rs:975]: Resolved.

File: src/engine/strat_engine/thinpool/thinpool.rs (hunk 2 above)
Comment (jbaublitz) [reply src/engine/strat_engine/thinpool/thinpool.rs:262]: Resolved.

File: src/engine/strat_engine/cmd.rs (hunk 1 above)
Comment (jbaublitz) [reply src/engine/strat_engine/cmd.rs]: Resolved.

**Review by mulkieran — APPROVED** [review]

**Pull request merged**